dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb

```diff
@@ -10,12 +10,12 @@ class WorkshopSurveyReportController < ReportControllerBase
     def workshop_survey_report
       survey_report = Hash.new
 
-      survey_report[:this_workshop] = get_score_for_workshops([@workshop], include_free_responses: true)
+      survey_report[:this_workshop] = get_score_for_workshops([@workshop], include_free_responses: true, facilitator_name_filter: facilitator_name_filter)
       all_my_workshops = (params[:organizer_view] ? Pd::Workshop.organized_by(current_user) : Pd::Workshop.facilitated_by(current_user)).where(course: @workshop.course).in_state(Pd::Workshop::STATE_ENDED)
-      survey_report[:all_my_workshops_for_course] = get_score_for_workshops(all_my_workshops)
+      survey_report[:all_my_workshops_for_course] = get_score_for_workshops(all_my_workshops, facilitator_name_filter: facilitator_name_filter)
 
       aggregate_for_all_workshops = JSON.parse(AWS::S3.download_from_bucket('pd-workshop-surveys', "aggregate-workshop-scores-#{CDO.rack_env}"))
-      survey_report[:all_workshops_for_course] = aggregate_for_all_workshops[@workshop.course].symbolize_keys
+      survey_report[:all_workshops_for_course] = aggregate_for_all_workshops[@workshop.course].try(&:symbolize_keys) || {}
```

Comment:
here too. This is a lot of logic in a controller. This should be in a model or lib class, split up into smaller units with tests. 

This looks like a bandaid for our known broken implementation, and from this I have little confidence we have fixed it.